modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1658,6 +1658,32 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 		}
 	}
 
+	if (p_function->is_vararg()) {
+		resolve_parameter(p_function->rest_parameter);
+		if (p_function->rest_parameter->datatype_specifier != nullptr) {
+			GDScriptParser::DataType specified_type = p_function->rest_parameter->get_datatype();
+			if (specified_type.kind != GDScriptParser::DataType::BUILTIN || specified_type.builtin_type != Variant::ARRAY ||
+					(specified_type.has_container_element_type(0) && !specified_type.get_container_element_type(0).is_variant())) {
```

Comment:
I feel like this should have a different message for typed arrays that says something like: "Typed arrays are currently not supported for the rest parameter.". Otherwise some users might open bugs thinking this is an oversight.